src/app/cart/cart.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Item } from '../models/item';
+import { CartService } from '../cart.service';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.css'],
+})
+
+export class CartComponent {
+  cart: Map<Item, number> = new Map();
+  constructor(private readonly cartService: CartService) {
+    this.cartService.getAllItems().subscribe((cart) => {
```

Comment:
Remember we should capture the subscription in order to unsubscribe when the component is destroyed.